### PR TITLE
refac: Modify CI checks for migrations scripts

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

- No longer a need to run .NET tests when delta migrations are modified
- Enable full all migrations script verifications for both hive and UC

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: .github/workflows/ci-databricks.yml [hunk 1]
```diff
@@ -55,19 +55,102 @@ jobs:
...
-          # Check if there are files in the primary directory
-          if [ "$(ls -A $PRIMARY_DIR)" ]; then
-              DIR_TO_USE=$PRIMARY_DIR
-          else
-              DIR_TO_USE=$SECONDARY_DIR
+          # Gets all files in migration_scripts folder on the main branch excluding the __init__ file and put them into at comma seperated string
+          SCRIPT_FILES_IN_MAIN=$(find ./source/databricks/calculation_engine/package/datamigration_hive/migration_scripts -type f -name '*.sql' -printf "%f,")
+          echo "::set-output name=script_files_in_main::$SCRIPT_FILES_IN_MAIN"
+
+      - name: Verify migration scripts
```
Comment (BjarkeMeier) [inline .github/workflows/ci-databricks.yml:62]: Document what's being verified? It's quite hard to figure out by reading this wall of script code.

Comment (JesperGomes) [reply .github/workflows/ci-databricks.yml:62]: done

File: .github/workflows/ci-databricks.yml [hunk 2]
```diff
@@ -55,23 +55,117 @@ jobs:
...
+        run: |
+          # This step verifies:
+          # 1. That all files in main branch exists in the PR branch TODO: remove this check when we the pipeline supports deleting scripts
```
Comment (BjarkeMeier) [inline .github/workflows/ci-databricks.yml:65]:
```suggestion
          # 1. That all files in main branch exists in the PR branch TODO JMG: remove this check when the pipeline supports deleting scripts
```

Comment (BjarkeMeier) [reply .github/workflows/ci-databricks.yml:65]: Comment applies in both steps.

File: .github/workflows/ci-databricks.yml [hunk 3]
```diff
@@ -55,23 +55,117 @@ jobs:
...
+          # 2. The timestamp of the latest script in the PR branch is greater than the latest script in the main branch
+          # 3. The timestamp of the latest script in the PR branch is less than the current datetime
+          # 4. The timestamp of the latest script in the PR branch has an underscore after the datetime
```
Comment (BjarkeMeier) [inline .github/workflows/ci-databricks.yml:68]: Does it also prevent the problem with too much precision (to many digits)?

Comment (JesperGomes) [reply .github/workflows/ci-databricks.yml:68]: I don't think so. Does it @johevemi ?

**Review by BjarkeMeier — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]

**Pull request merged**